backend/kernelCI_app/utils.py

```diff
@@ -111,9 +113,137 @@ class FilterParams:
+            "test.path": self._handle_path,
+            "boot.path": self._handle_path,
+        }
+
         self.filters = []
-        self.create_filters_from_req(request)
+        if process_body:
+            self.create_filters_from_body(data)
+        else:
+            self.create_filters_from_req(data)
+
+        self._processFilters()
+
+    def _handle_boot_status(self, current_filter: Dict):
+        self.filterBootStatus.add(current_filter["value"])
+
+    def _handle_boot_duration(self, current_filter: Dict):
+        value = current_filter["value"]
+        operation = current_filter["comparison_op"]
+        if operation == "lte":
+            self.filterBootDurationMax = toIntOrDefault(value, None)
+        else:
+            self.filterBootDurationMin = toIntOrDefault(value, None)
+
+    def _handle_test_status(self, current_filter: Dict):
+        self.filterTestStatus.add(current_filter["value"])
+
+    def _handle_test_duration(self, current_filter: Dict):
+        value = current_filter["value"]
+        operation = current_filter["comparison_op"]
+        if operation == "lte":
+            self.filterTestDurationMax = toIntOrDefault(value, None)
+        else:
+            self.filterTestDurationMin = toIntOrDefault(value, None)
+
+    def _handle_config_name(self, current_filter: Dict):
+        self.filterConfigs.add(current_filter["value"])
+
+    def _handle_compiler(self, current_filter: Dict):
+        self.filterCompiler.add(current_filter["value"])
+
+    def _handle_architecture(self, current_filter: Dict):
+        self.filterArchitecture.add(current_filter["value"])
+
+    def _handle_hardware(self, current_filter: Dict):
+        self.filterHardware.add(current_filter["value"])
+
+    def _handle_path(self, current_filter: Dict):
+        if current_filter["field"] == "boot.path":
+            self.filterBootPath = current_filter["value"]
+        else:
+            self.filterTestPath = current_filter["value"]
+
+    def _handle_build_valid(self, current_filter: Dict):
+        self.filterBuildValid.add(current_filter["value"])
+
+    def _handle_build_duration(self, current_filter: Dict):
+        value = current_filter["value"][0]
+        operation = current_filter["comparison_op"]
+        if operation == "lte":
+            self.filterBuildDurationMax = toIntOrDefault(value, None)
+        else:
+            self.filterBuildDurationMin = toIntOrDefault(value, None)
+
+    def _processFilters(self):
+        try:
+            for current_filter in self.filters:
+                field = current_filter["field"]
+                # Delegate to the appropriate handler based on the field
+                if field in self.filter_handlers:
+                    self.filter_handlers[field](current_filter)
+        except InvalidComparisonOP as e:
+            return HttpResponseBadRequest(getErrorResponseBody(str(e)))
+
+    def create_filters_from_body(self, body: Dict):
```

Comment:
one day we need to make this a little better but the improvements here are good for today